robolectric/src/main/java/org/robolectric/internal/BuckManifestFactory.java

```diff
@@ -0,0 +1,77 @@
+package org.robolectric.internal;
+
+import org.robolectric.annotation.Config;
+import org.robolectric.manifest.AndroidManifest;
+import org.robolectric.res.Fs;
+import org.robolectric.res.FsFile;
+import org.robolectric.res.ResourcePath;
+import org.robolectric.util.Logger;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.ListIterator;
+
+public class BuckManifestFactory implements ManifestFactory {
+
+  private static final String BUCK_ROBOLECTRIC_RES_DIRECTORIES = "buck.robolectric_res_directories";
+  private static final String BUCK_ROBOLECTRIC_ASSETS_DIRECTORIES = "buck.robolectric_assets_directories";
```

Comment:
Im ok with having build system agnostic system properties, but this is what the current buck build system passes in right now. I would like to maintain this right now for compatibility and robolectric could also additionally intercept generic system properties. Thoughts?